Helper/Data.php

```diff
@@ -178,24 +178,36 @@ public function subscribeEmailToKlaviyoList($email, $first_name=null, $last_name
         return $response;
     }
 
-    public function unsubscribeEmailFromKlaviyoList($email) 
+    public function unsubscribeEmailFromKlaviyoList($email)
     {
         $list_id = $this->getNewsletter();
         $api_key = $this->getPrivateApiKey();
 
-        $fields = [
-            'api_key=' . $api_key,
-            'email=' . urlencode($email),
-        ];
+        $url = 'https://a.klaviyo.com/api/v2/list/' . $list_id . '/subscribe';
+        $fields = json_encode([
+            'api_key' => (string)$api_key,
+            'emails' => [(string)$email],
+        ]);
 
-        $ch = curl_init();
-        curl_setopt($ch, CURLOPT_URL, 'https://a.klaviyo.com/api/v1/list/' . $list_id . '/members/exclude');
-        curl_setopt($ch, CURLOPT_POST, count($fields));
-        curl_setopt($ch, CURLOPT_POSTFIELDS, join('&', $fields));
-        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+        $curl = curl_init();
+        curl_setopt_array($curl, [
+            CURLOPT_URL => $url,
```

Comment:
this is okay and nothing wrong as it's just using v2... but i would much prefer a full refactor to create an isolated method that would handle making api requests, you would pass some arguments like `path, method, body` and then we wouldn't need to use curl all over the place like it currently is

Reply:
Fixed